### PR TITLE
Add docker build script to inject version information

### PR DESCRIPTION
**Pull request by StevenACoffman**
Partially Address #504 - (Dockerfile doesn't embed correct version) per https://github.com/tailscale/tailscale/issues/504#issuecomment-763733107

Signed-off-by: Steve Coffman <steve@khanacademy.org>

### CODE REVIEW

File: build_docker.sh [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# it does differently from `go build` is burn git commit and version
+# information into the binaries inside docker, so that we can track down user
+# issues.
+
+set -eu
+
+eval $(./version/version.sh)
+
+GOFLAGS='xversion -ldflags '"-X tailscale.com/version.Long=${VERSION_LONG} -X tailscale.com/version.Short=${VERSION_SHORT} -X tailscale.com/version.GitCommit=${VERSION_GIT_HASH}"
+docker build --build-arg goflags_arg="'""${GOFLAGS}""'" -t tailscale:tailscale .
```
Comment (bradfitz) [inline build_docker.sh:14]: can you add a newline at the end of this line, and also add a big fat warning banner to the top of this script like the Dockerfile itself has?

**Review by bradfitz — CHANGES_REQUESTED** [review]
Seems reasonable. @danderson, any thoughts?

File: build_docker.sh [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+# Also, see the various bugs tagged "containers":
+#    https://github.com/tailscale/tailscale/labels/containers
+#
+############################################################################
+
+set -eu
+
+eval $(./version/version.sh)
+
+GOFLAGS='xversion -ldflags '"-X tailscale.com/version.Long=${VERSION_LONG} -X tailscale.com/version.Short=${VERSION_SHORT} -X tailscale.com/version.GitCommit=${VERSION_GIT_HASH}"
```
Comment (danderson) [inline build_docker.sh:30]: I think you're missing `-tags` at the start of GOFLAGS, the first part should be `-tags xversion`.

**Review by danderson — APPROVED** [review]

File: build_docker.sh (hunk 2 above)
Comment (StevenACoffman) [reply build_docker.sh:30]: Oops! You're right. Fixing.

**Comment by danderson** [comment]
Thanks!

**Pull request merged**